packages/lesswrong/lib/sql/UpdateQuery.ts

```diff
@@ -163,7 +163,10 @@ class UpdateQuery<T extends DbObject> extends Query<T> {
     format: (resolvedField: string, updateValue: Atom<T>[]) => Atom<T>[],
   ): Atom<T>[] {
     try {
-      const updateValue = this.compileUpdateExpression(value);
+      const fieldType = this.getField(field);
+      const arrayValueInNonArrayJsonbField = fieldType && !fieldType.isArray() && fieldType.toConcrete() instanceof JsonType && Array.isArray(value);
```

Comment:
This is where most of the work happens.  In the case where we have an array-value in a `JSONB` (not `JSONB[]`) field, we need to pass in a `JsonType` to `Arg` so that it uses a `::JSBON` typehint rather than `::JSONB[]` (which causes an error).